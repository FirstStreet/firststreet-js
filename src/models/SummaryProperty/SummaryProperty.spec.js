const Property = require('./index');
const summaryMock = require('../../__mocks__/summary');

let summary = null;

describe('Property', () => {
  beforeEach(() => {
    summary = new Property(summaryMock);
  });

  it('should initalize', () => {
    expect(summary).toMatchSnapshot();
  });

  it('should have FSID', () => {
    expect(summary.FSID).toBeDefined();
  });

  it('should have primaryNumber', () => {
    expect(summary.primaryNumber).toBeDefined();
  });

  it('should have lastUpdated', () => {
    expect(summary.lastUpdated).toBeDefined();
  });

  it('should have streetName', () => {
    expect(summary.streetName).toBeDefined();
  });

  it('should have zipCode', () => {
    expect(summary.zipCode).toBeDefined();
  });

  it('should have city', () => {
    expect(summary.city).toBeDefined();
  });

  it('should have state', () => {
    expect(summary.state).toBeDefined();
  });

  it('should have polygon data', () => {
    expect(summary.polygon).toBeDefined();
  });

  it('should have polygon geoJSON', () => {
    expect(summary.polygonGeoJSON).toBeDefined();
  });

  it('should have center data', () => {
    expect(summary.center).toBeDefined();
  });

  it('should have center geoJSON', () => {
    expect(summary.centerGeoJSON).toBeDefined();
  });

  it('should have bounds data', () => {
    expect(summary.bounds).toBeDefined();
  });

  it('should have bounds geoJSON', () => {
    expect(summary.boundsGeoJSON).toBeDefined();
  });

  it('should have elevation', () => {
    expect(summary.elevation).toBeDefined();
  });

  it('should have femaZone', () => {
    expect(summary.femaZone).toBeDefined();
  });

  it('should have lotSize', () => {
    expect(summary.lotSize).toBeDefined();
  });

  it('should have countyFips', () => {
    expect(summary.countyFips).toBeDefined();
  });

  it('should have firstFloodRisk', () => {
    expect(summary.firstFloodRisk).toBeDefined();
  });

  it('should have floodRisks', () => {
    expect(summary.floodRisks).toBeDefined();
  });
});
