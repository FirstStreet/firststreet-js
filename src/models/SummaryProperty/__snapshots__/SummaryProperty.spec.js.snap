// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property should initalize 1`] = `
SummaryProperty {
  "data": Object {
    "FSID": 100032470544,
    "results": Object {
      "firstFloodRisk": Object {
        "floodID": "kt",
        "year": 2018,
      ***REMOVED***,
      "floodRisks": Array [
        Object {
          "floodID": Array [
            "em",
            "kt",
            "c5",
            "c3",
            "c1",
          ],
          "year": 2023,
        ***REMOVED***,
        Object {
          "floodID": Array [
            "em",
            "kt",
            "c5",
            "c3",
            "c1",
          ],
          "year": 2028,
        ***REMOVED***,
        Object {
          "floodID": Array [
            "em",
            "kt",
            "c5",
            "c3",
            "c1",
          ],
          "year": 2033,
        ***REMOVED***,
        Object {
          "floodID": Array [
            "em",
            "kt",
            "c5",
            "c3",
            "c1",
          ],
          "year": 2018,
        ***REMOVED***,
      ],
      "location": Object {
        "FSID": 100032470544,
        "city": Object {
          "FSID": 1222175,
          "Name": "Fernandina Beach",
        ***REMOVED***,
        "countyFips": 10003,
        "distance": 0,
        "elevation": 1085,
        "femaZone": null,
        "floorArea": 2228,
        "geometry": Object {
          "bounds": Object {
            "coordinates": Array [
              Array [
                Array [
                  -75.6456815054783,
                  39.4418499120952,
                ],
                Array [
                  -75.6451478217579,
                  39.4418499120952,
                ],
                Array [
                  -75.6451478217579,
                  39.4421122701503,
                ],
                Array [
                  -75.6456815054783,
                  39.4421122701503,
                ],
                Array [
                  -75.6456815054783,
                  39.4418499120952,
                ],
              ],
            ],
            "type": "Polygon",
          ***REMOVED***,
          "center": Object {
            "coordinates": Array [
              -75.6453718684964,
              39.4419892114799,
            ],
            "type": "Point",
          ***REMOVED***,
          "polygon": Object {
            "coordinates": Array [
              Array [
                Array [
                  Array [
                    -75.6451595133313,
                    39.4418499120952,
                  ],
                  Array [
                    -75.6451533032778,
                    39.4418767064764,
                  ],
                  Array [
                    -75.6451494009962,
                    39.441903761694,
                  ],
                  Array [
                    -75.6451478217579,
                    39.4419309570368,
                  ],
                  Array [
                    -75.6451485745461,
                    39.4419581780373,
                  ],
                  Array [
                    -75.6451516539709,
                    39.4419852949662,
                  ],
                  Array [
                    -75.6451570483542,
                    39.4420121947438,
                  ],
                  Array [
                    -75.6451647325431,
                    39.442038758741,
                  ],
                  Array [
                    -75.6451746733001,
                    39.4420648697161,
                  ],
                  Array [
                    -75.6451868239126,
                    39.442090409734,
                  ],
                  Array [
                    -75.6452018185913,
                    39.4421122701503,
                  ],
                  Array [
                    -75.645658659524,
                    39.4420479376122,
                  ],
                  Array [
                    -75.6456815054783,
                    39.4419626061836,
                  ],
                  Array [
                    -75.6451595133313,
                    39.4418499120952,
                  ],
                ],
              ],
            ],
            "type": "MultiPolygon",
          ***REMOVED***,
        ***REMOVED***,
        "landUse": "residential",
        "lastUpdated": "2019-06-04T15:55:08.566254Z",
        "lotSize": 861,
        "postdirection": "S",
        "predirection": null,
        "primaryNumber": "212",
        "state": "Delaware",
        "streetName": "APPOQUIN",
        "zipCode": 19734,
      ***REMOVED***,
    ***REMOVED***,
    "type": "property",
  ***REMOVED***,
  "location": Object {
    "FSID": 100032470544,
    "city": Object {
      "FSID": 1222175,
      "Name": "Fernandina Beach",
    ***REMOVED***,
    "countyFips": 10003,
    "distance": 0,
    "elevation": 1085,
    "femaZone": null,
    "floorArea": 2228,
    "geometry": Object {
      "bounds": Object {
        "coordinates": Array [
          Array [
            Array [
              -75.6456815054783,
              39.4418499120952,
            ],
            Array [
              -75.6451478217579,
              39.4418499120952,
            ],
            Array [
              -75.6451478217579,
              39.4421122701503,
            ],
            Array [
              -75.6456815054783,
              39.4421122701503,
            ],
            Array [
              -75.6456815054783,
              39.4418499120952,
            ],
          ],
        ],
        "type": "Polygon",
      ***REMOVED***,
      "center": Object {
        "coordinates": Array [
          -75.6453718684964,
          39.4419892114799,
        ],
        "type": "Point",
      ***REMOVED***,
      "polygon": Object {
        "coordinates": Array [
          Array [
            Array [
              Array [
                -75.6451595133313,
                39.4418499120952,
              ],
              Array [
                -75.6451533032778,
                39.4418767064764,
              ],
              Array [
                -75.6451494009962,
                39.441903761694,
              ],
              Array [
                -75.6451478217579,
                39.4419309570368,
              ],
              Array [
                -75.6451485745461,
                39.4419581780373,
              ],
              Array [
                -75.6451516539709,
                39.4419852949662,
              ],
              Array [
                -75.6451570483542,
                39.4420121947438,
              ],
              Array [
                -75.6451647325431,
                39.442038758741,
              ],
              Array [
                -75.6451746733001,
                39.4420648697161,
              ],
              Array [
                -75.6451868239126,
                39.442090409734,
              ],
              Array [
                -75.6452018185913,
                39.4421122701503,
              ],
              Array [
                -75.645658659524,
                39.4420479376122,
              ],
              Array [
                -75.6456815054783,
                39.4419626061836,
              ],
              Array [
                -75.6451595133313,
                39.4418499120952,
              ],
            ],
          ],
        ],
        "type": "MultiPolygon",
      ***REMOVED***,
    ***REMOVED***,
    "landUse": "residential",
    "lastUpdated": "2019-06-04T15:55:08.566254Z",
    "lotSize": 861,
    "postdirection": "S",
    "predirection": null,
    "primaryNumber": "212",
    "state": "Delaware",
    "streetName": "APPOQUIN",
    "zipCode": 19734,
  ***REMOVED***,
***REMOVED***
`;
